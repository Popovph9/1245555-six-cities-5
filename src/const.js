export const FLAT_TYPES = [`apartment`, `room`, `house`, `hotel`];

export const FLAT_OFFERS = [`Wifi`, `Heating`, `Kitchen`, `Cable TV`];

export const FLAT_PHOTOS = [`img/apartment-01.jpg`, `img/room.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`];

export const CITIES_NAMES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export const RATING_GRADES = [`5`, `4`, `3`, `2`, `1`];

export const RATING_TITLES = [`perfect`, `good`, `not bad`, `badly`, `terribly`];

export const PIN_URL = `/img/pin.svg`;

export const PIN_ACTIVE_URL = `/img/pin-active.svg`;

export const PIN_SIZES = [30, 30];

export const RENDERED_REVIEWS = 10;

export const RENDERED_PINS = 10;

export const MAX_RENDERED_PHOTOS = 6;

export const STAR_WIDTH = 20;

export const MIN_REVIEW_LENGTH = `50`;

export const MAX_REVIEW_LENGTH = `300`;

export const CITIES_CLASS = `cities__places-list tabs__content`;

export const PLACES_CLASS = `near-places__list`;

export const CITIES_SUBCLASS = `cities`;

export const PLACES_SUBCLASS = `near-places`;

export const EMPTY_PAGE_CLASSNAME = `page__main page__main--index page__main--index-empty`;

export const MAIN_PAGE_CLASSNAME = `page__main page__main--index`;

export const ACTIVE_CLASSNAME = `locations__item-link tabs__item tabs__item--active`;

export const CASUAL_CLASSNAME = `locations__item-link tabs__item`;

export const STAR_CLASSNAME = `property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro`;

export const COMMON_CLASSNAME = `property__avatar-wrapper user__avatar-wrapper`;

export const OPEN_CLASSNAME = `places__options--opened`;

export const DEFAULT_SORTING = `Popular`;

export const HIGHT_PRICE_SORTING = `Price: high to low`;

export const LOW_PRICE_SORTING = `Price: low to high`;

export const RATING_SORTING = `Top rated first`;

export const DEFAULT_CITY = CITIES_NAMES[0];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  ROOM: `/offer`,
  MAIN: `/`,
};

export const APIRoute = {
  HOTELS: `/hotels`,
  FAVORITES: `/favorite`,
  COMMENTS: `/comments`,
  NEAR: `/nearby`,
  LOGIN: `/login`,
};
