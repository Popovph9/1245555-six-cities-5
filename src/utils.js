import moment from "moment";

export const getRating = (rating, width) => {
  return Math.round(rating * width);
};

export const getReviewFormatDate = (date) => {
  return moment(date).format(`MMMM YYYY`);
};

export const getReviewDate = (date) => {
  return moment(date).format(`YYYY-MM-DD`);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
