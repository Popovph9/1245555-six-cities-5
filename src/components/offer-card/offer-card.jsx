import React from "react";
import {connect} from "react-redux";
import {setActivePin, resetActivePin, getCurrentOffer} from "../../store/action";
import PropTypes from "prop-types";
import {getRating} from "../../utils";
import {STAR_WIDTH} from "../../const";
import offersProp from "../../mocks/offers.prop";
import {PLACES_SUBCLASS} from "../../const";


const OfferCard = ({className, offers, onCardClick, setActivePinAction, resetActivePinAction, getCurrentOfferAction, onOfferClick}) => {

  const placesCardClass = `${className}__card place-card`;
  const citiesCardClass = `${className}__place-card place-card`;

  return (
    <React.Fragment>
      {offers.map((offer, i) => (
        <article
          key={`${i}`}
          className={className !== PLACES_SUBCLASS ? citiesCardClass : placesCardClass}
          onMouseEnter={() => {
            setActivePinAction(offer);
          }}
          onMouseLeave={resetActivePinAction}
          onClick={() => {
            onCardClick();
            getCurrentOfferAction(offer);
            onOfferClick(offer.id);
          }}
        >
          {offer.isPremium && className !== PLACES_SUBCLASS ?
            <div className="place-card__mark">
              <span>Premium</span>
            </div>
            : null}
          <div className={`${className}__image-wrapper place-card__image-wrapper`}>
            <a href="#">
              <img className="place-card__image" src={offer.picture} width="260" height="200" alt="Place image"/>
            </a>
          </div>
          <div className="place-card__info">
            <div className="place-card__price-wrapper">
              <div className="place-card__price">
                <b className="place-card__price-value">&euro;{offer.price}</b>
                <span className="place-card__price-text">&#47;&nbsp;night</span>
              </div>
              <button
                className={offer.isFavorite ? `place-card__bookmark-button--active button` : `place-card__bookmark-button button`} type="button">
                <svg className="place-card__bookmark-icon" width="18" height="19">
                  <use xlinkHref="#icon-bookmark"></use>
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="place-card__rating rating">
              <div className="place-card__stars rating__stars">
                <span style={{width: `${getRating(offer.rating, STAR_WIDTH)}%`}}></span>
                <span className="visually-hidden">Rating</span>
              </div>
            </div>
            <h2 className="place-card__name">
              <a href="#">{offer.headline}</a>
            </h2>
            <p className="place-card__type">{offer.type}</p>
          </div>
        </article>
      ))}
    </React.Fragment>
  );
};


OfferCard.propTypes = {
  className: PropTypes.string.isRequired,
  offers: offersProp,
  onCardClick: PropTypes.func.isRequired,
  setActivePinAction: PropTypes.func.isRequired,
  resetActivePinAction: PropTypes.func.isRequired,
  getCurrentOfferAction: PropTypes.func.isRequired,
  onOfferClick: PropTypes.func,
};

const mapDispatchToProps = (dispatch) => ({
  setActivePinAction(offer) {
    dispatch(setActivePin(offer));
  },
  resetActivePinAction() {
    dispatch(resetActivePin());
  },
  getCurrentOfferAction(offer) {
    dispatch(getCurrentOffer(offer));
  }
});

export {OfferCard};
export default connect(null, mapDispatchToProps)(OfferCard);
