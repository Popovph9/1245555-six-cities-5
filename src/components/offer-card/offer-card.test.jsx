import React from "react";
import renderer from "react-test-renderer";
import {OfferCard} from "./offer-card";
import {CITIES_SUBCLASS, PLACES_SUBCLASS, AuthorizationStatus} from "../../const";
import {noop} from "../../utils";
import {MOCK_OFFERS, MOCK_CURRENT_OFFER, MOCK_FAVORITE_OFFERS} from "../../test-data";

describe(`Should OfferCard render correctly`, () => {

  it(`Should OfferCard render correctly with main page class and unauthorized user`, () => {
    const OfferCardComponent = renderer.create(

        <OfferCard
          className={CITIES_SUBCLASS}
          offers={MOCK_OFFERS}
          onCardClick={noop}
          setActivePinAction={noop}
          resetActivePinAction={noop}
          refreshOfferList={noop}
          currentOfferId={MOCK_CURRENT_OFFER.id}
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          refreshNearOffersList={noop}
          changeFavoriteAction={noop}
          getCurrentOfferAction={noop}
          redirectToRouteAction={noop}
        />).toJSON();

    expect(OfferCardComponent).toMatchSnapshot();
  });

  it(`Should OfferCard render correctly with main page class and authorized user`, () => {
    const OfferCardComponent = renderer.create(

        <OfferCard
          className={CITIES_SUBCLASS}
          offers={MOCK_FAVORITE_OFFERS}
          onCardClick={noop}
          setActivePinAction={noop}
          resetActivePinAction={noop}
          refreshOfferList={noop}
          currentOfferId={MOCK_CURRENT_OFFER.id}
          authorizationStatus={AuthorizationStatus.AUTH}
          refreshNearOffersList={noop}
          changeFavoriteAction={noop}
          getCurrentOfferAction={noop}
          redirectToRouteAction={noop}
        />).toJSON();

    expect(OfferCardComponent).toMatchSnapshot();
  });

  it(`Should OfferCard render correctly with secondary page class and unauthorized user`, () => {
    const OfferCardComponent = renderer.create(

        <OfferCard
          className={PLACES_SUBCLASS}
          offers={MOCK_OFFERS}
          onCardClick={noop}
          setActivePinAction={noop}
          resetActivePinAction={noop}
          refreshOfferList={noop}
          currentOfferId={MOCK_CURRENT_OFFER.id}
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          refreshNearOffersList={noop}
          changeFavoriteAction={noop}
          getCurrentOfferAction={noop}
          redirectToRouteAction={noop}
        />).toJSON();

    expect(OfferCardComponent).toMatchSnapshot();
  });

  it(`Should OfferCard render correctly with secondary page class and authorized user`, () => {
    const OfferCardComponent = renderer.create(

        <OfferCard
          className={PLACES_SUBCLASS}
          offers={MOCK_FAVORITE_OFFERS}
          onCardClick={noop}
          setActivePinAction={noop}
          resetActivePinAction={noop}
          refreshOfferList={noop}
          currentOfferId={MOCK_CURRENT_OFFER.id}
          authorizationStatus={AuthorizationStatus.AUTH}
          refreshNearOffersList={noop}
          changeFavoriteAction={noop}
          getCurrentOfferAction={noop}
          redirectToRouteAction={noop}
        />).toJSON();

    expect(OfferCardComponent).toMatchSnapshot();
  });
});
