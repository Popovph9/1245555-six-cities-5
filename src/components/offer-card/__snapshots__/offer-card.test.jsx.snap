// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferCard render correctly Should OfferCard render correctly with main page class and authorized user 1`] = `
Array [
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            323
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "86%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
]
`;

exports[`Should OfferCard render correctly Should OfferCard render correctly with main page class and unauthorized user 1`] = `
Array [
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            323
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "86%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            232
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          House in countryside
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
]
`;

exports[`Should OfferCard render correctly Should OfferCard render correctly with secondary page class and authorized user 1`] = `
Array [
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            323
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "86%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
]
`;

exports[`Should OfferCard render correctly Should OfferCard render correctly with secondary page class and unauthorized user 1`] = `
Array [
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            323
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "86%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            232
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="#"
        >
          House in countryside
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
]
`;
